### DISCUSSION TITLE
How should I be running tests locally?

### DISCUSSION THREAD

**Discussion by boothmanrylan** [google/Xee#98] [Q&A] [answered]
Since #94 I have not been able to run tests locally.

Previously I ran `pytest` and the tests all ran as expected.

Now when I run `pytest`, all of the ext integration tests fail with `KeyError: GOOGLE_APPLICATION_CREDENTIALS`

I've tried running `earthengine authenticate` before running `pytest` but get the same error.

I've also tried running the following before `pytest`:
```shell
earthengine authenticate
export GOOGLE_APPLICATION_CREDENTIALS=/path/to/file/earth/engine/authenticate/saved/credentials/to.json
```

Doing this causes all the ext integration tests to fail with `google.auth.exceptions.MalformedError: Missing credential_source. A 'file' or 'url' must be provided.`

If I revert this: https://github.com/google/Xee/blob/07b8676498c3bd2f9dc10ec7010a9e125e313f6c/xee/ext_integration_test.py#L44-L48

Back to this:
```python
def init_ee_for_tests():
  ee.Initialize(opt_url='https://earthengine-highvolume.googleapis.com')
```
Everything works as expected.

Any suggestions would be appreciated thanks.

**Comment by naschmitz** [comment, marked as answer]
Hi Rylan, you should be able to run integration tests with `USE_ADC_CREDENTIALS=1 python -m unittest xee/ext_integration_test.py`. See https://github.com/google/Xee?tab=readme-ov-file#how-to-run-integration-tests for more details.
